core/controllers/library.py

```diff
@@ -67,13 +67,21 @@ class LibraryPage(base.BaseHandler):
 
     def get(self):
         """Handles GET requests."""
+        search_mode = (self.request.route.name == 'library_search_page')
```

Comment:
Actually, there's an issue underway to get rid of the route names, in order to make main.py more comprehensible. Can we instead check whether self.request.uri contains 'search'?